components/locid/src/extensions/other/mod.rs

```diff
@@ -100,6 +104,18 @@ impl Other {
     pub fn get_ext(&self) -> char {
         self.0 .0 as char
     }
+
+    pub(crate) fn for_each_subtag_str<E, F>(&self, f: &mut F) -> Result<(), E>
```

Comment:
note: with this signature it's not possible to compare two iterators. maybe that's fine and we don't need cmp_bytes, but that's a major downside of external iteration

Reply:
I re-wrote cmp_bytes in this PR to do the comparison manually.

It's a little awkward, but up to 65% faster than the iterator spaghetti

Reply:
Ah sorry, I misspoke: I meant that "maybe it's fine and we _only_ need cmp_bytes" but basically I was saying that we won't be able to compare this locale with a different locale representation. Perhaps that's fine!

Reply:
Ok I see your point. I'm inclined to go with "maybe that's fine", since this is internal code and we can refactor it later if needed

Reply:
Sounds good! We can always try the iterator approach if we need it later.